code/modules/mod/modules/module_pathfinder.dm

```diff
@@ -0,0 +1,238 @@
+	if(!length(path)) //Cannot reach target. Give up and announce the issue.
+		to_chat(H, "<span class='warning'>No viable path found!</span>")
+		return FALSE
+	to_chat(H, "<span class='notice>Suit on route!</span>")
+	animate(module.mod, 0.2 SECONDS, pixel_x = 0, pixel_y = 0)
+	module.mod.add_overlay(jet_icon)
+	RegisterSignal(module.mod, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+	mod_move(target)
+	return TRUE
+
+/obj/item/implant/mod/proc/set_path(list/newpath)
+	if(newpath == null)
+		end_recall(FALSE)
+	path = newpath ? newpath : list()
+
+/obj/item/implant/mod/proc/end_recall(successful = TRUE)
+	if(!module?.mod)
+		return
+	module.mod.cut_overlay(jet_icon)
+	module.mod.transform = matrix()
+	UnregisterSignal(module.mod, COMSIG_MOVABLE_MOVED)
+	if(!successful)
+		to_chat(imp_in, "<span class='warning'>Lost connection to suit!</span>")
+		path = list() //Stopping endless end_recall with luck.
+
+/obj/item/implant/mod/proc/on_move(atom/movable/source, atom/old_loc, dir, forced)
+	SIGNAL_HANDLER
+
+	var/matrix/mod_matrix = matrix()
+	mod_matrix.Turn(get_angle(source, imp_in))
+	source.transform = mod_matrix
+
+/obj/item/implant/mod/proc/mod_move(dest)
+	dest = get_turf(dest) //We must always compare turfs, so get the turf of the dest var if dest was originally something else.
+	if(get_turf(module.mod) == dest) //We have arrived, no need to move again.
+		for(var/mob/living/carbon/human/H in range(1, module.mod))
+			if(H == imp_in)
+				module.attach(imp_in)
+				end_recall()
+				return TRUE
+		end_recall(FALSE)
+		return FALSE
+
+
+	if(!dest || !path || !length(path)) //A-star failed or a path/destination was not set.
+		set_path(null)
+		return FALSE
+
+	var/turf/last_node = get_turf(path[length(path)]) //This is the turf at the end of the path, it should be equal to dest.
+	if(dest != last_node) //The path should lead us to our given destination. If this is not true, we must stop.
+		set_path(null)
+		return FALSE
+
+	var/step_count = 3 //Temp speed for now
+
+	if(step_count >= 1 && tries < 5)
+		for(var/step_number in 1 to step_count)
+			// Hopefully this wont fill the buckets too much
+			addtimer(CALLBACK(src, PROC_REF(mod_step)), 2 * (step_number - 1))
+	if(tries >= 5)
+		set_path(null)
+	var/target = get_turf(imp_in)
+	var/mob/living/carbon/human/H = imp_in
+	set_path(get_path_to(module.mod, target, 150, id = H.wear_id, simulated_only = FALSE)) //Yes, science proves jetpacks work in space. More at 11.
+	addtimer(CALLBACK(src, PROC_REF(mod_move), target), 6) //I'll value this properly soon
+
+	return TRUE
+
+/obj/item/implant/mod/proc/mod_step() //Step,increase tries if failed
+	if(!length(path))
+		return FALSE
+	for(var/obj/machinery/door/D in range(2, module.mod))
+		if(D.operating || D.emagged)
+			continue
+		if(D.requiresID() && D.allowed(imp_in))
+			if(D.density)
+				D.open()
+
+
+	if(!step_towards(module.mod, path[1]))
```

Comment:
An "list index out of bounds" did happen here earlier today.

Reply:
Weird because that *should* be caught by the `!length(path)` check above.

Reply:
We see elsewhere `if(!path || !length(path))` maybe that can help?

Reply:
Oh yeah good catch.